File: internal/module/captcha/model/captcha.go
package model

type Point struct {
	X int `json:"x"`
	Y int `json:"y"`
}

type CaptchaGenerateResponse struct {
	Token    string `json:"token"`
	Bg       string `json:"bg"`
	BgWidth  int    `json:"bgWidth"`
	BgHeight int    `json:"bgHeight"`
	Chars    string `json:"chars"`
}

type CaptchaVerifyRequest struct {
	Token  string  `json:"token" binding:"required"`
	Points []Point `json:"points" binding:"required"`
}

type CaptchaVerifyResponse struct {
	Success bool   `json:"success"`
	Token   string `json:"token"`
	Message string `json:"message"`
}

File: web/src/api/captcha.ts
import request from './index'
import type { Result } from './index'

export interface CaptchaGenerateResponse {
  token: string
  bg: string
  bgWidth: number
  bgHeight: number
  chars: string
}

export interface CaptchaPoint {
  x: number
  y: number
}

export interface CaptchaVerifyResponse {
  success: boolean
  token: string
  message: string
}

export function getCaptcha() {
  return request.get<any, Result<CaptchaGenerateResponse>>('/captcha/generate')
}

export function verifyCaptcha(data: { token: string; points: CaptchaPoint[] }) {
  return request.post<any, Result<CaptchaVerifyResponse>>('/captcha/verify', data)
}
